Ckan_API/Ckan_API.py

```diff
@@ -0,0 +1,37 @@
+from urllib.parse import urljoin
+
+import requests
+
+
+class CkanAPICall:
+
+    def __init__(self, ckan_api_url, ckan_api_key):
+        self.ckan_api_url = ckan_api_url
+        self.ckan_api_key = ckan_api_key
+
+    def get_all_datasets(self):
+        packages = self.package_list()
+
+        datasets = []
+        for package in packages:
+            datasets.append(self.package_show(package))
```

Comment:
Not critical but when building a list, the typical Pythonic structure is to use a list comprehension:

```suggestion
        datasets = [self.package_show(p) for p in packages]
```
It's a bit nicer than for loops because the person reading can immediately see that what we are doing here is forming a list from a different iterable, while for loops could be used for many things. Again not critical and mostly a matter of style, but for many Python developers this is considered a preferred style. You can also use `map` here which would also support the same idiom (mapping one iterable to another).

Reply:
One more thought: as this may very well be thousands of datasets, it may be wise to return a generator here and not a list. This could be done using Python's amazing generator expressions:

```python
    def get_all_datasets(self):
        packages = self.package_list()
        return (self.package_show(p) for p in packages)
```
The difference being the use of parenthesis instead of square brackets. 

This would mean that at no point you hold the full list of (converted) packages in memory, but on each iteration over this function's return value, `package_show(p)` will be called.

Reply:
you are a great mentor BTW 😄 ! Thanks